### ISSUE TITLE
[REVIEW]: ropenblas: Download, Compile and Link OpenBLAS Library with R

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2769]
**Submitting author:** @prdm0 (<a href="http://orcid.org/0000-0003-1591-8300">Pedro Rafael Marinho</a>)
**Repository:** <a href="https://github.com/prdm0/ropenblas" target ="_blank">https://github.com/prdm0/ropenblas</a>
**Version:** v0.2.9
**Editor:** @timtroendle
**Reviewers:** @pratikvn, @myousefi2016
**Archive:** <a href="https://doi.org/10.5281/zenodo.4618251" target="_blank">10.5281/zenodo.4618251</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/6903d5f01c23e13a6ce28d84af287406/status.svg)](https://joss.theoj.org/papers/6903d5f01c23e13a6ce28d84af287406)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/6903d5f01c23e13a6ce28d84af287406"><img src="https://joss.theoj.org/papers/6903d5f01c23e13a6ce28d84af287406/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/6903d5f01c23e13a6ce28d84af287406/status.svg)](https://joss.theoj.org/papers/6903d5f01c23e13a6ce28d84af287406)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/prdm0/ropenblas" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@pratikvn & @sahilseth, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @timtroendle know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/prdm0/ropenblas">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@prdm0) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/prdm0/ropenblas">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@prdm0) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @pratikvn, @sahilseth it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
Prezado @timtroendle , Is there a maximum time limit for improvements to be implemented? When the suggestions are implemented, should I generate a new PDF using `@whedon generate pdf`?

Kind regards.

**Comment by timtroendle** [comment]
Dear @prdm0, there is no hard time limit for you to implement changes. However, if this should take you a longer time it would be great if you could let us know. At the moment we are waiting for review comments from @sahilseth.

**Comment by prdm0** [comment]
Dear @timtroendle, I intend to quickly make the changes suggested by the first reviewer. When they are implemented, should I update the PDF or should I wait for suggestions from the second reviewer?

**Comment by timtroendle** [comment]
That should be fine as long as it doesn't disturb the second reviewer. @sahilseth, please let us know in case you prefer to see the original paper.

**Comment by timtroendle** [comment]
@sahilseth, can you please give us an update where you stand with your review?

**Comment by prdm0** [comment]
@timtroendle, I'm waiting for some feedback from @sahilseth so that I can implement all suggestions for improvement from both reviewers. 

Kind regards.

**Comment by sahilseth** [comment]
@timtroendle Could you please re-add me as a reviewer, I am not able to submit feedback. Sorry, we couldn't find that repository invitation.

**Comment by timtroendle** [comment]
Sure. Not sure why you weren't assigned, sorry about that. Did this work?

**Comment by sahilseth** [comment]
I was still getting the same error if you would like o re-add me as a reviewer, I will try again. If not, I can copy-paste the review as a comment here tonight.

**Comment by timtroendle** [comment]
@whedon add @sahilseth  as reviewer

**Comment by whedon** [comment]
OK, @sahilseth  is now a reviewer

**Comment by timtroendle** [comment]
I gave it another try. Please let me know if it works now.

**Comment by sahilseth** [comment]
Perfect, that worked

**Comment by timtroendle** [comment]
@sahilseth, can you please give us an update on where you stand with your review? We need to finish this first round of reviews. Can you please let me know how long this will take you?

**Comment by timtroendle** [comment]
@whedon add @jgoldfar as reviewer

**Comment by whedon** [comment]
OK, @jgoldfar is now a reviewer

**Comment by timtroendle** [comment]
@openjournals/dev, sorry for the trouble but Whedon did not assign the new reviewer to this issue and I am unable to assign them manually. Is there anything I can do here?

**Comment by arfon** [comment]
@whedon re-invite @sahilseth as reviewer

**Comment by whedon** [comment]
@sahilseth already has access.

**Comment by arfon** [comment]
@whedon re-invite @jgoldfar as reviewer

**Comment by whedon** [comment]
The reviewer already has a pending invite.

@jgoldfar please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by arfon** [comment]
> @openjournals/dev, sorry for the trouble but Whedon did not assign the new reviewer to this issue and I am unable to assign them manually. Is there anything I can do here?

@timtroendle - I'm not sure exactly who has the issue here but @sahilseth should be able to complete their review (update checkboxes etc.) fine, @jgoldfar needs to accept the invite in the message above before they will be able to update their checkboxes.

@timtroendle - just as an FYI - adding reviewers after the pre-review stage is generally a little more challenging (as you're finding out!) which is why we prefer to have the reviewers assigned in the pre-review issue if at all possible.

**Comment by timtroendle** [comment]
@arfon - great, thanks for the update and for letting me know.

**Comment by timtroendle** [comment]
@sahilseth, @jgoldfar, please can you give us an update about how much time you will need to finish the review?

**Comment by prdm0** [comment]
@Pratikvn sent me suggestions. Is another reviewer missing to send the corrections?

**Comment by timtroendle** [comment]
Yes, we are waiting for a second review to be finished.

**Comment by jgoldfar** [comment]
I will complete my review (and the checklist above - unable to update on mobile) by Friday. Thanks for the reminder!

**Comment by prdm0** [comment]
I received @pratikvn reviews on the PROBLEM of the ropenblas project <https://github.com/prdm0/ropenblas/issues/21>. @jgoldfar will also send in the same way or will something here also be sent?

**Comment by timtroendle** [comment]
Hi @jgoldfar, did you have the time to look at this submission?

**Comment by timtroendle** [comment]
@sahilseth @jgoldfar We need to proceed. Please let me know whether you can finish your review within the next two weeks or whether you prefer to drop the review.

**Comment by timtroendle** [comment]
@whedon add @myousefi2016 as reviewer.

**Comment by whedon** [comment]
OK, @myousefi2016 is now a reviewer

**Comment by myousefi2016** [comment]
Hi @timtroendle, thanks for giving me this opportunity to review [ropenblas](https://github.com/prdm0/ropenblas). One quick comment for @prdm0 to speed up the review process a bit: The software paper is great and describes very interesting and useful details. I just reflected that in my evaluation in the **Software paper** section. But, for _Quality of writing_: I suggest to double check with an native English speaker (I'm not a native English speaker either) to make sure there is no typo or grammatical error to improve the level of writing quality.

**Comment by prdm0** [comment]
Dear @myousefi2016 , thanks for the suggestion. I will revise the text as suggested. I look forward to the other suggestions, if any, in [**issues**](https://github.com/prdm0/ropenblas/issues) of the package project in the Github. Kind regards.

**Comment by myousefi2016** [comment]
@timtroendle I think I'm done with the review and I'm just waiting for revision of the software paper.

**Comment by prdm0** [comment]
Dear, soon I will be returning the PDF with all the requested corrections. Best regards.

**Comment by prdm0** [comment]
Dear Editors,

I proceeded with the changes suggested by the reviewers. I modified the documentation on GitHub and the library website as suggested by one of the reviewers, as well as made several changes to the paper. Among them I mention the inclusion of a benchmark, changes in writing and a better explanation of some concepts.

@whedon

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
Dear, I did one more complete review of the article's English. I believe that the article is now in a presentable format. Thank you for all the suggestions that have enriched the material.

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by timtroendle** [comment]
@whedon remove @sahilseth as reviewer

**Comment by whedon** [comment]
OK, @jgoldfar is no longer a reviewer

**Comment by timtroendle** [comment]
Thank you for the update, @prdm0. @pratikvn and @myousefi2016, could you please have a look at the revisions and update your corresponding checklist above? Also, a review issue is still open: https://github.com/prdm0/ropenblas/issues/21. @pratikvn, can you please close the issue if the changes have been implemented sufficiently?

**Comment by pratikvn** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by pratikvn** [comment]
@timtroendle , Everything looks good to me modulo some final paper changes which I have provided on https://github.com/prdm0/ropenblas/issues/21 . 

After these changes have been implemented, I can close that issue and we can proceed here.

**Comment by prdm0** [comment]
@pratikvn, obrigado pelas sugestões. Irei proceder com as melhorias e em breve gerarei um novo PDF.

**Comment by prdm0** [comment]
Dear @pratikvn and @timtroendle, I made all the requested changes. @pratikvn I made the corrections to the text you requested, I eliminated unnecessary parts, as indicated in [prdm0/ropenblas#21](https://github.com/prdm0/ropenbla/issues/21). I also improved the benchmarks for building a presentable table. The benchmark plot now includes two algebraic routines for comparison. The new PDF of the paper can be found in the comment below.

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by pratikvn** [comment]
@timtroendle, the paper looks good from my side.

**Comment by timtroendle** [comment]
Thank you @pratikvn.

**Comment by prdm0** [comment]
@timtroendle , Do I have to do anything else for the paper to be finalized? Best regards.

**Comment by timtroendle** [comment]
Not at this point, thanks @prdm0 .

**Comment by timtroendle** [comment]
@myousefi2016, did you have a chance to look at the updates?

**Comment by myousefi2016** [comment]
@timtroendle Yes, I checked and it looks good to me as well.

**Comment by prdm0** [comment]
Dear @timtroendle, how long does it take for the paper to be listed on the JOSS page? Best regards.

**Comment by timtroendle** [comment]
@prdm0, as both reviewers are satisfied with the changes, we can start the acceptance process, which should not take long. At this point could you please do the following:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

**Comment by prdm0** [comment]
@whedon set v0.2.9 as version

**Comment by whedon** [comment]
I'm sorry @prdm0, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by prdm0** [comment]
Dear @timtroendle ,

1. [https://github.com/prdm0/ropenblas/tree/v0.2.9](https://github.com/prdm0/ropenblas/tree/v0.2.9), **tag v0.2.9**;
2. [https://zenodo.org/record/4618251#.YFOGiytv83w](https://zenodo.org/record/4618251#.YFOGiytv83w);
3. Done metadata;
4. [https://doi.org/10.5281/zenodo.4618251](https://doi.org/10.5281/zenodo.4618251), **DOI**: 10.5281/zenodo.4618251

**Comment by prdm0** [comment]
> I'm sorry @prdm0, I'm afraid I can't do that. That's something only editors are allowed to do.

My language is not English. I must have run an unallowed command.

**Comment by prdm0** [comment]
Dear @timtroendle ,

1 - <https://github.com/prdm0/ropenblas/tree/v0.2.9>, **tag v0.2.9**;
2 - <https://zenodo.org/record/4618251#.YFOGiytv83w>;
3 - Done metadata;
4 - <https://doi.org/10.5281/zenodo.4618251>, **DOI**: [10.5281/zenodo.4618251](https://doi.org/10.5281/zenodo.4618251).

**Comment by timtroendle** [comment]
Dear @prdm0, thanks for this! Your submission is almost ready for publication. However, the paper needs more work. With more than 2000 words and 6 pages, it’s too long. Can you please shorten the paper to about 1000 words or less? Here are two possibilities:

First, the paper contains a lot of information taken from the documentation. This information is not required in the JOSS article. This concerns in particular the installation instructions and the function overview in the “Brief explanation” section. You could shorten this section drastically be limiting it to the most important function(s), or by discussing an example use only.

Second, the paper contains redundancy: the sections “Summary”, “Statement of Need”, and “Introduction” have quite some overlap. If you streamline these sections, you should be able to cut the number of words significantly. You could, for example, discuss the importance of computational efficiency and OpenBLAS in the Summary section; explain the benefits of ropenblas in the “Statement of Need” section; and reveal further details of ropenblas in a third section (likely not called “Introduction”). This is only a suggestion and I leave the details up to you.

**Comment by prdm0** [comment]
Dear @timtroendle , I will proceed with the changes now.

**Comment by prdm0** [comment]
Dear @timtroendle, I made the reductions so that there was a reduction of one page. I believe that it complies with the other papers in JOSS that have up to a maximum of 5 pages.

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
Dear @timtroendle, it decreases even more and the paper, The material now has less than 2000 words, more precisely with 1998.

```
[prdm0@samsung paper]$ pdftotext paper.pdf - | wc -w
1998
```

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
@timtroendle, when possible, let me know according to the program you use to count words if the PDF above is suitable. I made a comparison with the latest publications, considering my way of telling, the paper has fewer words than those last publications.

I reduced it by a little more than one page. I believe that the paper is now suitable. 

Best regards.

**Comment by timtroendle** [comment]
Thank you for the offer @prdm0, but this is not about the exact number of words. Rather, it's about having the right content in the article. Let me have a look at your most recent changes and come back to you soon.

**Comment by timtroendle** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by timtroendle** [comment]
@prdm0, I made some edits to your draft (attached). Can you please work these edits in?
[10.21105.joss.02769-4-TT.pdf](https://github.com/openjournals/joss-reviews/files/6182863/10.21105.joss.02769-4-TT.pdf)

I want to highlight the "Brief explanation" section here. As I mentioned above, large parts of this section are taken from the documentation and provide a level of detail that is not necessary for a JOSS article. Can you please remove or shorten this section? There is no need to discuss individual functions of ropenblas in this article.

**Comment by prdm0** [comment]
@timtroendle  I'll see how to do it. In the section Bifef explanation there is everything that was asked by the reviewers. But I can remove it.

**Comment by prdm0** [comment]
I will remove all explanations of the functions and leave only the benchmark.

**Comment by prdm0** [comment]
@timtroendle I made all the modifications. The PDF with the modified paper is below.

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by prdm0** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02769/joss.02769/10.21105.joss.02769.pdf) :page_facing_up: :point_left:

**Comment by myousefi2016** [comment]
Hi @prdm0, just a quick comment here: please finalize all your changes to the manuscript before generating pdf. It's not necessary to do some incremental changes and then generate the pdf. I see there are lots of generate pdf commands here which makes this issue a bit lengthy and hard to follow. It is possible to avoid this by finalizing all your changes and just have one generate pdf command at the end.

**Comment by prdm0** [comment]
> Hi @prdm0, just a quick comment here: please finalize all your changes to the manuscript before generating pdf. It's not necessary to do some incremental changes and then generate the pdf. I see there are lots of generate pdf commands here which makes this issue a bit lengthy and hard to follow. It is possible to avoid this by finalizing all your changes and just have one generate pdf command at the end.

Dear @myousefi2016 I know that. However, the PDF was in a format here in RStudio and when the graphic was generated it was on another page. That's why it was generated. But don't worry, I know that and the last generation has been realized and is above your comment.

**Comment by prdm0** [comment]
In summary, here the PDF was in one form and the robot generated it a little differently. It has been corrected and the final version is above.

**Comment by timtroendle** [comment]
Thank you @prdm0. At this point, we can move forward with the acceptance process.

**Comment by timtroendle** [comment]
@whedon set 10.5281/zenodo.4618251 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4618251" target="_blank">10.5281/zenodo.4618251</a> is the archive.

**Comment by timtroendle** [comment]
@whedon set v0.2.9 as version

**Comment by whedon** [comment]
OK. v0.2.9 is the version.

**Comment by timtroendle** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2158

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2158, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by prdm0** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @prdm0, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by prdm0** [comment]
> I'm sorry @prdm0, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

@timtroendle , Pardon me. I thought that I should run the command since the tag had changed to recommended paper for acceptance. I believe that now nothing else depends on me. If you need anything, I'm at your disposal.

**Comment by prdm0** [comment]
@timtroendle, do I need to proceed with anything else or all the steps from now on just depend on other people? Best regards.

**Comment by timtroendle** [comment]
Nothing you need to do at this point, @prdm0. The editor in chief will take this forward. Please give us some time to handle the final steps.

**Comment by kthyng** [comment]
@openjournals/dev Two names of the author are missing in the XML file — does this mean the paper won't have the correct author name?

**Comment by kthyng** [comment]
@prdm0 I am looking through your paper and it needs quite a bit of editing. I started working on it in this [PR](https://github.com/prdm0/ropenblas/pull/28) but I changed sentences substantially to try to reflect my understanding of your writing. Please make sure I have done so accurately before merging. 

Quite a bit more editing is necessary. I couldn't understand line 42 in the .md file to try to edit it. There is also a lot of repetition. For example, the fact that the Linux distribution doesn't matter and that behavior in the package is the same was mentioned at least 3 times in the first 1.5 pages. Please read through the manuscript carefully and edit it. The recommended work count for the paper is 250-1000 words so no need to add anything that isn't necessary.

**Comment by prdm0** [comment]
> > @openjournals/dev Two names of the author are missing in the XML file — does this mean the paper won't have the correct author name?
> 
> @prdm0 - what should your `surname` be here? https://github.com/openjournals/joss-papers/pull/2158/files#diff-93ddcb6dc29bed45dbfd3d925b04a9e6f5f6932c1205907d7e70ab16fe204d20R40

Submeti as modificações. 

Name: Pedro Rafael
Last name: Diniz Marinho

**Comment by arfon** [comment]
:+1: thanks @prdm0. @kthyng - we'll need to manually publish this one (I can do this) as Whedon can't currently handle name/lastname combinations like this yet sorry.

**Comment by kthyng** [comment]
@arfon has taken care of publishing, so congrats on your new publication @prdm0! Thanks to editor @timtroendle and reviewers @pratikvn and @myousefi2016 for your time, hard work, and expertise!!

**Comment by timtroendle** [comment]
Congratulation @prdm0! Thanks everyone; thank you @pratikvn and @myousefi2016 for your contributions!
